### ISSUE TITLE
Bug in saving the best weights

### ISSUE THREAD

**Issue by ArminSchenk** [citoverse/cito#56]
In the trainings function where the best weights are saved the if clause is wrong:
if(min(model$losses$valid_l[epoch-1]) > model$losses$valid_l[epoch])

```
model$losses$valid_l[1:(epoch-1)] would be correct.
```
I combined the "saving best weights" part with the "early stopping" part to avoid redundancy in calculating the best loss:

```
    # Save best weights
    if(model$training_properties$validation != 0 & !is.null(valid_dl)) {
      if(model$losses$valid_l[epoch] < best_val_loss) {
        best_val_loss = model$losses$valid_l[epoch]
        model$weights[[1]] =  lapply(model$net$parameters,function(x) torch::as_array(x$to(device="cpu")))
        counter = 0
      }
    } else {
      if(model$losses$train_l[epoch] < best_train_loss) {
        best_train_loss = model$losses$train_l[epoch]
        model$weights[[1]] =  lapply(model$net$parameters,function(x) torch::as_array(x$to(device="cpu")))
        counter = 0
      }
    }

    ### early stopping ###
    if(is.numeric(model$training_properties$early_stopping)) {
      if(counter >= model$training_properties$early_stopping) {
        break
      }
      counter = counter + 1
    }
```
